common/services/app/google-analytics.tsx

```diff
@@ -42,18 +43,38 @@ function createToggleString(toggles: Toggles | undefined): string | null {
     : null;
 }
 
-export const Ga4DataLayer: FunctionComponent<Props> = ({ data }) => {
-  const toggleString = createToggleString(data.toggles);
+export const Ga4DataLayer: FunctionComponent<Props> = ({
+  data,
+  hasAnalyticsConsent,
+}) => {
+  const abTestsToggleString = createABToggleString(data.toggles);
 
-  return toggleString ? (
+  return data.toggles?.cookiesWork?.value || abTestsToggleString ? (
     <script
       dangerouslySetInnerHTML={{
         __html: `
-        window.dataLayer = window.dataLayer || [];
-        window.dataLayer.push({
-          toggles: '${toggleString}'
-        });
-        `,
+            window.dataLayer = window.dataLayer || [];
+
+            ${
+              data.toggles?.cookiesWork?.value
+                ? `function gtag(){window.dataLayer.push(arguments);}
+              
+              gtag('consent', 'default', {
```

Comment:
I think that we need some more fields here to adhere to "consent mode v2": https://developers.google.com/tag-platform/security/guides/consent?consentmode=advanced#upgrade-consent-v2, specifically `ad_user_data` and `ad_personalization`.

I expect we'll just hard-code the values to `denied` though.

Reply:
Yess I wasn't sure we used those because adding them [caused me TS errors](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/node_modules/@types/gtag.js/index.d.ts), looks like the gtag helpers don't know about them?

```
No overload matches this call.
  The last overload gave the following error.
    Object literal may only specify known properties, and 'ad_personalization' does not exist in type 'ConsentParams'.
```
<img width="440" alt="Screenshot 2024-03-08 at 11 48 54" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/110461050/3fc4469d-60b0-4c81-9b1f-1ed5436dad5d">

That being said, as I'm looking at this, it's only a problem in the TS code when we `update` the values, and not on the `default`/initialisation one. And we should only change the `analytics_storage` value if they allow analytics and the others should be `denied` from the get go, including `ad_storage`, which I currently mark as `granted` if analytics is. So I'll change that to add everything to the consent initialisation and only change `analytics_storage` on consent given.

If ever we want those to be changeable, we'll deal with TS errors then. Not a problem for now. I'll stop thinking out loud and apply changes 👍